### PR TITLE
Fixed GPU documentation for HF pipelines

### LINKED ISSUE #580: Unable to use GPU accelerated Optimum Onnx transformer model for inference
### System Info

```shell
Optimum Version: 1.5.0
Ubuntu 20.04 Linux 
Python version 3.8
```

### Who can help?

@JingyaHuang @echarlaix 
When following the documentation on https://huggingface.co/docs/optimum/main/en/onnxruntime/usage_guides/gpu for 1.5.0 version optimum. We get the following error:

---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
[<ipython-input-7-8429fcab1e09>](https://localhost:8080/#) in <module>
     19             "education",
     20             "music"]
---> 21 pred = onnx_z0(sequence_to_classify, candidate_labels, multi_class=False)

8 frames
[/usr/local/lib/python3.8/dist-packages/onnxruntime/capi/onnxruntime_inference_collection.py](https://localhost:8080/#) in bind_input(self, name, device_type, device_id, element_type, shape, buffer_ptr)
    454         :param buffer_ptr: memory pointer to input data
    455         """
--> 456         self._iobinding.bind_input(
    457             name,
    458             C.OrtDevice(

RuntimeError: Error when binding input: There's no data transfer registered for copying tensors from Device:[DeviceType:1 MemoryType:0 DeviceId:0] to Device:[DeviceType:0 MemoryType:0 DeviceId:0]

This is reproducible on google colab gpu instance as well. This is observed from 1.5.0 version only and 1.4.1 works as expected.

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

!pip install optimum[onnxruntime-gpu]==1.5.1
!pip install transformers onnx

from optimum.onnxruntime import ORTModelForSequenceClassification

ort_model = ORTModelForSequenceClassification.from_pretrained(
    "philschmid/tiny-bert-sst2-distilled",
    from_transformers=True,
    provider="CUDAExecutionProvider",
)

from optimum.pipelines import pipeline
from transformers import AutoTokenizer

tokenizer = AutoTokenizer.from_pretrained("philschmid/tiny-bert-sst2-distilled")

pipe = pipeline(task="text-classification", model=ort_model, tokenizer=tokenizer)
result = pipe("Both the music and visual were astounding, not to mention the actors performance.")
print(result)

### Expected behavior

Inference fails due to device error, which is not expected.

### PR DESCRIPTION
**Pull request by smiraldr**
Fixes #580
Added a note for the user to initialise HF Pipelines with device parameter which is required for GPU Execution.

### CODE REVIEW

File: docs/source/onnxruntime/usage_guides/gpu.mdx [hunk 1]
```diff
@@ -69,14 +69,15 @@ ort_model = ORTModelForSequenceClassification.from_pretrained(
 ```
 
 The model can then be used with the common 🤗 Transformers API for inference and evaluation, such as [pipelines](https://huggingface.co/docs/optimum/onnxruntime/usage_guides/pipelines):
+Note that Transformers pipeline expects it to be initialized with device for GPU accelerated execution as well.
```
Comment (fxmarty) [inline docs/source/onnxruntime/usage_guides/gpu.mdx:72]:
```suggestion
When using Transformers pipeline, note that the `device` argument should be set to perform pre- and post-processing on GPU.
```

**Review by fxmarty — APPROVED** [review]
Just left a small comment but otherwise it's good to merge!

File: docs/source/onnxruntime/usage_guides/gpu.mdx [hunk 2]
```diff
@@ -69,14 +69,15 @@ ort_model = ORTModelForSequenceClassification.from_pretrained(
 ```
 
 The model can then be used with the common 🤗 Transformers API for inference and evaluation, such as [pipelines](https://huggingface.co/docs/optimum/onnxruntime/usage_guides/pipelines):
```
Comment (fxmarty) [inline docs/source/onnxruntime/usage_guides/gpu.mdx:71]:
```suggestion
The model can then be used with the common 🤗 Transformers API for inference and evaluation, such as [pipelines](https://huggingface.co/docs/optimum/onnxruntime/usage_guides/pipelines).
```

**Comment by smiraldr** [comment]
Made the needed changes ! looks good to merge.

**Comment by fxmarty** [comment]
Awesome thanks a lot for improving the doc!

**Pull request merged**